be/src/olap/rowset/segment_v2/inverted_index_writer.cpp

```diff
@@ -126,6 +126,12 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
         }
     }
 
+    void close_on_error() override {
+        if (_index_writer) {
+            _index_writer->close();
```

Comment:
`close()` will write idx file and temporary directory. These files should be deleted too.